xls/fuzzer/value_generator.cc

```diff
@@ -193,20 +195,40 @@ absl::StatusOr<Expr*> GenerateDslxConstant(absl::BitGenRef bit_gen,
                                            Module* module,
                                            TypeAnnotation* type) {
   dslx::Span fake_span = dslx::FakeSpan();
-  if (auto* builtin_type = dynamic_cast<dslx::BuiltinTypeAnnotation*>(type);
-      builtin_type != nullptr) {
+
+  if (std::optional<dslx::BitVectorMetadata> metadata =
+          dslx::ExtractBitVectorMetadata(type);
+      metadata.has_value()) {
+    auto bit_count_or = absl::visit(
+        xls::Visitor{
+            [&](int64_t bit_count) -> absl::StatusOr<int64_t> {
+              return bit_count;
+            },
+            [&](Expr* expr) -> absl::StatusOr<int64_t> {
+              absl::StatusOr<int64_t> bit_count_or = EvaluateDimExpr(expr);
+              if (bit_count_or.ok()) {
```

Comment:
It is still in the XLS style guide to name it `thing_or` -- I'd ask that it be changed in the style guide if it's the new official policy:

> Prefer using XLS_ASSIGN_OR_RETURN / XLS_RETURN_IF_ERROR when appropriate, but when binding a StatusOr wrapped value prefer to name it thing_or so that it can be referenced without the wrapper as thing; e.g.
-- https://google.github.io/xls/xls_style/

Reply:
Sounds good, no strong feelings from me, just wanted to make sure there's consistent guidance.

Reply:
Also note that it looks like there's like 86 instances of `absl::StatusOr<.*?> \w+_or` so probably would be a good gardening task to clean those up.

Reply:
I guess just in terms of "the downsides for the rationale section" it's easier to forget to check the `.status()` if there's not a sigil indicating you should. For example:

```
absl::StatusOr<MyThing> my_thing = ...
my_thing->stuff  // whoops
```

vs 

```
absl::StatusOr<MyThing> my_thing_or = ...
my_thing_or->stuff  // it's an _or, did I remember to check it was non-error via e.g. XLS_RETURN_IF_ERROR
```

But probably doesn't matter too much in practice on account of test coverage showing any forgotten test.

Reply:
No more `_or` values in the PR now.

Reply:
> Also note that it looks like there's like 86 instances of `absl::StatusOr<.*?> \w+_or` so probably would be a good gardening task to clean those up.

Yep, @hzeller brought this up. I will start working on this, time permitting, though as I mentioned in my response to him, it's good to follow the idea of "converge on new styles over time". 

> it's easier to forget to check the `.status()` if there's not a sigil indicating you should. For example:

I checked and unfortunately, it doesn't look like our OSS clang-tidy checks have a flag for this. Internally, we have such a clang-tidy check for "unchecked-statusor-access".

Reply:
I see. Sounds useful -- if you don't mind filing a tracking bug for OSSing it, such that others can collaborate on abseil-based repos more effectively, I'd appreciate that.

Reply:
> I see. Sounds useful -- if you don't mind filing a tracking bug for OSSing it, such that others can collaborate on abseil-based repos more effectively, I'd appreciate that.

I filed an internal bug about upstreaming this check since it's easier to point to what I'm talking about.